language-design.md

```diff
@@ -38,18 +44,17 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 ---
  
 **Question**
 
-
 In what way is an API a language? 
 
 **Response**
 
-
+Languages are collections of vocabulary and rules [Steele 1998]. We use languages to describe specific domains: natural languages describe the world around us, programming languages describe the domain in which we want a computer to perform tasks. Mernik describes an Application Programming Interface (API) as “a domain specific vocabulary of class, method, and function names” [Mernik et al., 2005]. We agree. This falls under our categorization of language because the API gives us all the tools necessary to describe the domain of the application. Additionally, APIs share several qualities of what we understand to be a well-designed language. For example, Bloch describes several must-haves in a good API, which include consistent naming conventions, ease of use, and self-documentation. When describing API naming conventions, Bloch explicitly says, “Strive for intelligibility, consistency, and symmetry. Every API is a little language, and people must learn to read and write it” [Bloch, 2006]. As in any well-designed language, API naming conventions call for a strong working vocabulary such that the users need make only a minimal effort to use it. Bloch explains how, “it should rarely require documentation to read code written to a good API. In fact, it should rarely require documentation to write it” [Bloch, 2006]. The API, like any other good language should make sense and be intuitive to use for anyone with some experience in it.
```

Comment:
I agree for the most part with the ways you've described in which an API is like a language and I guess, based on Steele's definition of a language, an API is a language. When I answered this question, I was thinking back to the Fowler reading, which replaces "rules" in the definition with "syntax." Certainly a syntax is a set of rules, but I don't think the rules of most APIs are entirely like a syntax. For the most part, functions and methods in an API don't really combine with each other in interesting ways (maybe I've only seen boring APIs, though), so they don't seem very syntactic to me. I don't know whether you had this in mind at all, but if you did I'd be curious to hear your thoughts on why APIs are languages by both definitions.